pkg/agent/monitortool/latency_store.go

```diff
@@ -0,0 +1,271 @@
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+
+	// Delete from the node IP map
+	delete(l.nodeTargetIPsMap, old.Name)
+
+	// Node name will not be changed in the same Node update operation.
+	l.updateNodeMap(new)
+}
+
+// updateNodeMap updates the nodeTargetIPsMap with the IPs of the given Node.
+func (l *LatencyStore) updateNodeMap(node *corev1.Node) {
+	nodeIPs, err := l.getNodeIPs(node)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get IPs for Node", "nodeName", node.Name)
+		return
+	}
+
+	l.nodeTargetIPsMap[node.Name] = nodeIPs
+}
+
+// getNodeIPs returns the target IPs of the given Node based on the agent mode.
+func (l *LatencyStore) getNodeIPs(node *corev1.Node) ([]net.IP, error) {
+	if l.isNetworkPolicyOnly {
+		transportIPs, err := getTransportIPs(node)
+		if err != nil {
+			return nil, err
+		}
+
+		return transportIPs, nil
+	} else {
+		gw0IPs, err := getGWIPs(node)
+		if err != nil {
+			return nil, err
+		}
+
+		return gw0IPs, nil
+	}
+}
+
+// getTransportIPs returns the transport IPs of the given Node.
+func getTransportIPs(node *corev1.Node) ([]net.IP, error) {
+	var transportIPs []net.IP
+	ips, err := k8s.GetNodeTransportAddrs(node)
+	if err != nil {
+		return transportIPs, err
+	}
+
+	if ips.IPv4 != nil {
+		transportIPs = append(transportIPs, ips.IPv4)
+	}
+	if ips.IPv6 != nil {
+		transportIPs = append(transportIPs, ips.IPv6)
+	}
+
+	return transportIPs, nil
+}
+
+// getGWIPs returns the gateway IPs of the given Node.
+func getGWIPs(node *corev1.Node) ([]net.IP, error) {
+	var gwIPs []net.IP
+
+	podCIDRStrs := getPodCIDRsOnNode(node)
+	if len(podCIDRStrs) == 0 {
+		// Skip the Node if it does not have a PodCIDR.
+		err := errors.New("node does not have a PodCIDR")
+		klog.ErrorS(err, "Node does not have a PodCIDR", "Node name", node.Name)
+		return gwIPs, err
+	}
+
+	// the 0th entry must match the podCIDR field. It may contain at most 1 value for
+	// each of IPv4 and IPv6.
+	// Both podCIDRStrs need to be parsed to get the gateway IP.
+	for _, podCIDR := range podCIDRStrs {
+		if podCIDR == "" {
+			err := errors.New("PodCIDR is empty")
+			klog.ErrorS(err, "PodCIDR is empty", "Node name", node.Name)
+			return gwIPs, err
+		}
+
+		peerPodCIDRAddr, _, err := net.ParseCIDR(podCIDR)
+		if err != nil {
+			klog.ErrorS(err, "Failed to parse PodCIDR", "Node name", node.Name)
+			return gwIPs, err
+		}
+
+		// Add first IP in CIDR to the map
+		peerGatewayIP := ip.NextIP(peerPodCIDRAddr)
+
+		// Only add the IP if it is an IPv4 or IPv6 address.
+		if peerGatewayIP.To4() != nil || peerGatewayIP.To16() != nil {
+			gwIPs = append(gwIPs, peerGatewayIP)
+		}
+	}
+
+	return gwIPs, nil
+}
+
+// getPodCIDRsOnNode returns the PodCIDRs of the given Node.
+func getPodCIDRsOnNode(node *corev1.Node) []string {
+	if node.Spec.PodCIDRs != nil {
+		return node.Spec.PodCIDRs
+	}
+
+	if node.Spec.PodCIDR == "" {
+		// Does not help to return an error and trigger controller retries.
```

Comment:
This comment was copied from node_route_controller.go, but it doesn't make sense in your case because you don't have a workqueue, so there is no retry mechanism. You should remove it.

Reply:
Thanks, I have removed them.